libraries/chain/block_log.cpp

```diff
@@ -615,6 +668,18 @@ namespace eosio { namespace chain {
             FC_LOG_AND_RETHROW()
          }
 
+         std::vector<char> read_serialized_block_by_num(uint32_t block_num) final {
+            try {
+               block_pos_size_t pos_size = get_block_position_and_size(block_num);
```

Comment:
I feel it would be cleaner with the following (and also `get_block_position_and_size` could return a `std::pair`)
```suggestion
               auto [position, size] = get_block_position_and_size(block_num);
```

Reply:
I think struct provides stronger type check, especially when those two members are of the same type, and it is more clear in return statement `return block_pos_size_t {.position = pos, .size = block_size}` than just return `{pos, block_size}`.

Reply:
I can still use structure binding to make it cleaner. Thanks.